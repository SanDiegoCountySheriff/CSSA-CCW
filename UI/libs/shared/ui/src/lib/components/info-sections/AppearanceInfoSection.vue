<template>
  <v-container
    fluid
    class="appearance-info-container rounded mt-3"
  >
    <v-banner
      single-line
      class="sub-header font-weight-bold text-xl text-left mb-5"
    >
      {{ $t(' Physical Appearance Information ') }}
      <template #actions>
        <v-btn
          v-if="
            applicationStore.completeApplication.application.status ==
            ApplicationStatus.Incomplete
          "
          icon
          @click="handleEditRequest"
        >
          <v-icon :color="$vuetify.theme.dark ? 'info' : 'info'">
            mdi-square-edit-outline
          </v-icon>
        </v-btn>
      </template>
    </v-banner>
    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Height Feet: ') }}
          </strong>
          {{ props.appearanceInfo.heightFeet }}
        </v-banner>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Height Inches: ') }}
          </strong>
          {{ props.appearanceInfo.heightInch }}
        </v-banner>
      </v-col>
    </v-row>
    <v-row>
      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Weight (lbs): ') }}
          </strong>
          {{ props.appearanceInfo.weight }}
        </v-banner>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Eye Color: ') }}
          </strong>
          {{ props.appearanceInfo.eyeColor }}
        </v-banner>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Hair Color: ') }}
          </strong>
          {{ props.appearanceInfo.hairColor }}
        </v-banner>
      </v-col>

      <v-col
        cols="12"
        lg="6"
      >
        <v-banner
          rounded
          class="text-left"
          single-line
        >
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Gender: ') }}
          </strong>
          {{ props.appearanceInfo.gender }}
        </v-banner>
      </v-col>
    </v-row>

    <v-row>
      <v-col>
        <v-banner class="text-left">
          <v-icon
            left
            color="primary"
          >
            mdi-account
          </v-icon>
          <strong>
            {{ $t('Physical Description: ') }}
          </strong>
          {{ props.appearanceInfo.physicalDesc }}
        </v-banner>
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup lang="ts">
import { ApplicationStatus } from '@shared-utils/types/defaultTypes'
import { useCompleteApplicationStore } from '@shared-ui/stores/completeApplication'
import { AppearanceInfoType } from '@shared-utils/types/defaultTypes'
import { useRouter } from 'vue-router/composables'

interface IAppearanceInfoSectionProps {
  appearanceInfo: AppearanceInfoType
  color: string
}

const props = defineProps<IAppearanceInfoSectionProps>()
const router = useRouter()
const applicationStore = useCompleteApplicationStore()

function handleEditRequest() {
  applicationStore.completeApplication.application.currentStep = 4
  router.push({
    path: '/form',
    query: {
      applicationId: applicationStore.completeApplication.id,
      isComplete: applicationStore.completeApplication.application.isComplete,
    },
  })
}
</script>

<style lang="scss" scoped>
.appearance-info-container {
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
}
</style>
