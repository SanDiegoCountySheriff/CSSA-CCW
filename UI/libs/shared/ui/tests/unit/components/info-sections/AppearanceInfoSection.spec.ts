/* eslint-disable @typescript-eslint/ban-ts-comment */

import AppearanceInfoSection from '@shared-ui/components/info-sections/AppearanceInfoSection.vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()

describe('AppearanceInfoSection', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
    //@ts-ignore
    wrapper = mount(AppearanceInfoSection, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        appearanceInfo: {
          gender: 'male',
          heightFeet: 5,
          heightInch: 5,
          weight: 5,
          hairColor: 'black',
          eyeColor: 'black',
          physicalDesc: 'something',
        },
      },
    })
  })

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
