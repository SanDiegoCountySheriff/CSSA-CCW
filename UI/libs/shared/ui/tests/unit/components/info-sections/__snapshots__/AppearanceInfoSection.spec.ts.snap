// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppearanceInfoSection Should match the snapshot 1`] = `
<div class="container appearance-info-container rounded container--fluid">
  <transition-stub name="expand-transition" class="font-weight-bold text-xl text-left mb-5">
    <div class="v-banner v-sheet theme--light v-banner--is-mobile">
      <div class="v-banner__wrapper">
        <div class="v-banner__content">
          <div class="v-banner__text">
            Physical Appearance Information
          </div>
        </div>
      </div>
    </div>
  </transition-stub>
  <div class="row mb-5">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Height Feet
        </span></span>
        <div class="info-text">
          5
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
          Height Inch
        </span></span>
        <div class="info-text">
          5
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Weight 
        </span></span>
        <div class="info-text">
          5
        </div>
      </div>
    </div>
  </div>
  <div class="row mb-5">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Hair Color 
        </span></span>
        <div class="info-text">
          black
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Eye Color 
        </span></span>
        <div class="info-text">
          black
        </div>
      </div>
    </div>
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Gender
        </span></span>
        <div class="info-text">
          male
        </div>
      </div>
    </div>
  </div>
  <div class="row">
    <div class="col-lg-6 col-12">
      <div class="info-row"><span class="font-weight-bold v-chip v-chip--label v-chip--no-color theme--light v-size--default"><span class="v-chip__content">
           Physical Description
        </span></span>
        <div class="info-text">
          something
        </div>
      </div>
    </div>
  </div>
</div>
`;
